apps/backend/src/libs/modules/mapbox/libs/types/get-mapbox-route-response-dto.type.ts

```diff
@@ -0,0 +1,39 @@
+import { type PointGeometry } from "~/libs/types/types.js";
+
+type GetMapboxRouteResponseDto = {
+	code: string;
+	routes: MapboxRoute[];
+	uuid: string;
+	waypoints: MapboxWaypoint[];
+};
+
+type MapboxLegs = {
+	admins: {
+		iso_3166_1: string;
+		iso_3166_1_alpha3: string;
+	}[];
+	distance: number;
+	duration: number;
+	summary: string;
+	weight: number;
+};
+
+type MapboxRoute = {
+	distance: number;
+	duration: number;
+	geometry: {
+		coordinates: PointGeometry["coordinates"][];
+		type: string;
+	};
+	legs: MapboxLegs[];
+	weight: number;
+	weight_name: string;
+};
+
+type MapboxWaypoint = {
+	distance: number;
+	location: PointGeometry["coordinates"][];
+	name: string;
+};
```

Comment:
Let's remove unnecessary fields and interfaces 
1. waypoints are deprecated, and we don't need them
2. We don't need legs for now, and not sure if we will. Also, we don't need weight
3. We only need to select the first recommendation per route. For this, we also need to pass `alternatives: false` to optimize the request
4. Eventually, before returning anything from this service, we need to remap it and return in the format and only the info we need 

```suggestion
type RouteContructResponseDto = { // if in shared
	route: {
	  distance: number;
	  duration: number;
	  geometry: LineStringGeometry
	}
	internalId: string; // uuid
};
```

As for geometry, I believe it should be of type
```
 {
    type: "LineString";  
    coordinates: PointGeometry["coordinates"][];
  };
```

If so, let's create a LineStringGeometry common type, similarly to PointGeometry